TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumerTest.java

```diff
@@ -315,6 +316,90 @@ public void testMetricCountsFor_testThatConnectionsAreKeptAliveAndShared(boolean
+                new TransformationLoader().getTransformerFactoryLoader(null), null);
+
+            var clientConnectionPool =  new ClientConnectionPool(testServer.localhostEndpoint(), sslContext,
+                "targetPool for testReadTimeoutHandler_responseTakesLongerThanTimeout", 1);
+
+            var timeShifter = new TimeShifter();
+            timeShifter.setFirstTimestamp(Instant.now());
+
+            var sendingFactory = new ReplayEngine(
+                new RequestSenderOrchestrator(clientConnectionPool,
+                    (replaySession, ctx1) -> new NettyPacketToHttpConsumer(replaySession, ctx1, responseTimeout)),
+                new TestFlowController(), timeShifter);
+
+            var ctx = rootContext.getTestConnectionRequestContext("TEST", 0);
+            var requestFinishFuture = TrafficReplayerTopLevel.transformAndSendRequest(transformingHttpHandlerFactory,
+                sendingFactory, ctx, Instant.now(), Instant.now(),
+                () -> Stream.of(EXPECTED_REQUEST_STRING.getBytes(StandardCharsets.UTF_8)));
+            // Make sure finalize is triggered shortly after the response timeout
+            var aggregatedResponse = requestFinishFuture.get(responseTimeout.plus(testDurationBuffer));
+            Assertions.assertInstanceOf(ReadTimeoutException.class, aggregatedResponse.getError());
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({"false", "true"})
+    public void testTimeBetweenRequestsLongerThanResponseTimeout(boolean useTls) throws Exception {
+        var testDurationBuffer = Duration.ofSeconds(10);
+        var responseTimeout = Duration.ofMillis(100);
+        var timeBetweenRequests = responseTimeout.plus(Duration.ofMillis(10));
+        try (var testServer = SimpleNettyHttpServer.makeServer(useTls,
+            NettyPacketToHttpConsumerTest::makeResponseContext)) {
+            var sslContext = !testServer.localhostEndpoint().getScheme().equalsIgnoreCase("https") ? null :
+                SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build();
+            var transformingHttpHandlerFactory = new PacketToTransformingHttpHandlerFactory(
+                new TransformationLoader().getTransformerFactoryLoader(null), null);
+
+            var clientConnectionPool = new ClientConnectionPool(testServer.localhostEndpoint(), sslContext,
+                "targetPool for testTimeBetweenRequestsLongerThanResponseTimeout", 1);
+
+            var timeShifter = new TimeShifter();
+            timeShifter.setFirstTimestamp(Instant.now());
+            var sendingFactory = new ReplayEngine(
+                new RequestSenderOrchestrator(clientConnectionPool,
+                    (replaySession, ctx1) -> new NettyPacketToHttpConsumer(replaySession, ctx1, responseTimeout)),
+                new TestFlowController(), timeShifter);
+            for (int i = 0; i < 2; ++i) {
+                if (i != 0) {
```

Comment:
nit, you could put this at the end of a while(true) loop after your i<2 test to break.